api/capi/src/nnstreamer-capi-single.c

```diff
@@ -343,18 +345,24 @@ ml_single_set_info_in_handle (ml_single_h single, gboolean is_input,
     configured = klass->output_configured (single_h->filter);
   }
 
-  if (tensors_info) {
-    if (!configured)
-      ml_single_set_inout_tensors_info (filter_obj, is_input, tensors_info);
-    ml_tensors_info_clone (dest, tensors_info);
-  } else {
-    ml_tensors_info_h info;
-
+  if (configured) {
+    /* get configured info and compare with input info */
     ml_single_get_tensors_info_from_filter (single_h->filter, is_input, &info);
+
+    if (tensors_info && !ml_tensors_info_is_equal (tensors_info, info, equal)) {
```

Comment:
For correctness you need to update the macro. the return of "is_equal" is not "is_equal'.